modules/core/src/main/java/org/apache/ignite/internal/util/ByteUtils.java

```diff
@@ -74,6 +85,17 @@ public static byte[] longToBytes(long l) {
         return putLongToBytes(l, new byte[8], 0);
     }
 
+    /**
+     * Converts a primitive {@code long} value to a byte array than can be compared with {@link ByteBuffer#compareTo(ByteBuffer)} with the
+     * same order as the original {@code long}.
+     *
+     * @param l Long value.
+     * @return Array of bytes.
+     */
+    public static byte[] longToComparableBytes(long l) {
```

Comment:
`Comparable` in the name is a bit confusing, it's not clear, comparable to what and why the alternative is not comparable (as it is comparable, but *differently*).

How about `longToBytesKeepingComparability()`? Or `longToBytesKeepingOrder()`